drivers/hmis/app/graphql/mutations/ce/calculate_client_eligibility.rb

```diff
@@ -0,0 +1,84 @@
+    argument :values_by_link_id, Types::JsonObject, required: true
+
+    field :project_types, [Types::HmisSchema::Enums::ProjectType], null: false
+
+    def resolve(enrollment_id:, form_definition_identifier:, values_by_link_id:)
+      enrollment = Hmis::Hud::Enrollment.viewable_by(current_user).find(enrollment_id)
+      access_denied! unless current_user.can_edit_enrollments_for?(enrollment)
+
+      # The match engine expects destination clients
+      client = enrollment.client.destination_client
+      return { project_types: [] } unless client
+
+      # Convert form values to field value overrides for CE evaluation
+      overrides = build_overrides(values_by_link_id, form_definition_identifier)
+
+      # Get evaluate client eligibility against all candidate pools
+      eligible_project_types = calculate_eligible_project_types(client, overrides)
+
+      { project_types: eligible_project_types }
+    end
+
+    private
+
+    def build_overrides(values_by_link_id, form_definition_identifier)
+      # Get the form definition to map from link_id to custom_field_key
+      form_definition = Hmis::Form::Definition.published.find_by(identifier: form_definition_identifier)
+      return {} unless form_definition
+
+      # Build overrides hash
+      values_by_link_id.filter_map do |link_id, value|
+        form_item = form_definition.link_id_item_hash[link_id]
+        custom_field_key = form_item&.dig('mapping', 'custom_field_key')
+        next unless custom_field_key
+
+        # Map to CDE field format for CE evaluation - see FieldMap
+        ["cde.custom_assessment.#{custom_field_key}", value]
+      end.to_h
+    end
+
+    def calculate_eligible_project_types(client, field_value_overrides)
+      field_map = Hmis::Ce::Match::Expression::FieldMap.new(current_date: Date.current)
+      eligible_pools = []
+
+      # Evaluate client against active candidate pools
+      Hmis::Ce::Match::CandidatePool.active.find_each do |pool|
+        clients = GrdaWarehouse::Hud::Client.where(id: client.id)
+        evaluator = Hmis::Ce::Match::Internal::ClientPoolEvaluator.new(clients, pool, field_map)
+        result = evaluator.call(client, field_value_overrides: field_value_overrides)
+
+        eligible_pools << pool unless result.failed? # result fails if client is ineligible
+      end
+
+      # Find projects that use the eligible pools and collect their project types
+      project_types = find_project_types_from_pools(eligible_pools)
+      project_types.uniq
+    end
+
+    def find_project_types_from_pools(pools)
+      return [] if pools.empty?
+
+      # Get project types from Unit Groups that use these pools.
+      # (Unit Groups and not Opportunities, because we want to return results about what projects the client is eligible for, regardless of availability)
+      Hmis::UnitGroup.
+        joins(:project).
+        where(candidate_pool_id: pools.map(&:id)).
+        pluck('Project.project_type').
+        compact.uniq
```

Comment:
I think we'll need with_ce_waitlists_enabled, once https://github.com/greenriver/hmis-warehouse/pull/5715  merged

Maybe better to pluck of off the project model, I'm not sure how pluck 'Project.project_type'  works as it's not quoted correctly
```suggestion
      Hmis::Hud::Project.
        with_ce_waitlists_enabled.
        joins(:unit_groups).
        merge(Hmis::UnitGroup.where(candidate_pool_id: pools.map(&:id)))
        distinct.
        pluck(:project_type)
```

Reply:
Thanks I like your improvement! For my education,
> as it's not quoted correctly

what does this mean?

Reply:
oh sorry- I was just suprised 'Project.project_type' works.

My understanding is postgres folds identifiers to lowercase unless you quote them. So `select Project.project_type from Project` is the same as `select project.project_type from project` which would be an error. Because we use case-sensitive identifiers, we'd actually want it to be `select "Project"."ProjectType" from "Project"` 

ActiveRecord does this translation/quoting for us usually. Maybe it's smart enough to handle 'Project.project_type' too

Reply:
I see thanks!